msprime/simulations.py

```diff
@@ -1236,19 +1236,19 @@ def __repr__(self):
 def matrix_exponential(A):
     """
     Returns the matrix exponential of A.
-
     https://en.wikipedia.org/wiki/Matrix_exponential
-
     Note: this is not a general purpose method and is only intended for use within
     msprime.
     """
```

Comment:
Could we use this version? I don't think the rounding or `Q` are actually doing anything.
```
    d, Y = np.linalg.eig(A)
    Yinv = np.linalg.pinv(Y)
    D = np.diag(np.exp(d))
    B = np.matmul(Y, np.matmul(D, Yinv))
    return B
```

Reply:
and the `asmatrix( )` calls are just making `*` do `matmul`, but less obviously.